File: problems/tier2-intermediate/016-amazon-locker/boilerplate/java/part1/Guided.java
import java.util.*;

// Data class (given).
class LockerOp {
    public String kind;
    public String s1;
    public String s2;
    public int i1;
    public int i2;

    public LockerOp(String kind, String s1, String s2, int i1, int i2) {
        this.kind = kind;
        this.s1 = s1;
        this.s2 = s2;
        this.i1 = i1;
        this.i2 = i2;
    }

    public LockerOp(String kind) {
        this(kind, "", "", 0, 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> locker_simulate(List<LockerOp> ops) {
        // TODO: write your solution
        return null;
    }

}

File: problems/tier2-intermediate/016-amazon-locker/boilerplate/cpp/part2/guided.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;


// Data class (given).
struct LockerOp {
    string kind;
    string s1;
    string s2;
    int i1;
    int i2;
    LockerOp(const string& kind_, const string& s1_ = "", const string& s2_ = "", int i1_ = 0, int i2_ = 0)
      : kind(kind_), s1(s1_), s2(s2_), i1(i1_), i2(i2_) {}
};

// HINT: introduce an abstraction so new variants don't change existing code.
// HINT: keep each piece small — one responsibility per class.

// HINT: start from what this must return, then work backwards to the state it needs.
vector<string> locker_simulate(vector<LockerOp> ops) {
    // TODO: write your solution
    return {};
}
